lib/authentication/authentication.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:admin_dashboard/authentication/authentication_service.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:github_sign_in/github_sign_in.dart';
+
+class FirebaseAuthenticationService implements AuthenticationService{
+  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
+
```

Comment:
I assume this service would be injected later to several classes, can we use Riverpod for DI and state management? and make use you provide this to the widget properly?